assets/dev/js/editor/regions/panel/pages/elements/elements.js

```diff
@@ -160,6 +172,28 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
 		this.categoriesCollection = categoriesCollection;
 	},
 
+	shouldNotAddWidget( widget ) {
+		const isContainerActive = elementorCommon.config.experimentalFeatures.container;
+
+		return ! widget.show_in_panel || ( 'inner-section' === widget.name && isContainerActive );
+	},
+
+	deepMerge( originalObj, replacementObj ) {
+		const mergedObj = { ...originalObj };
+
+		for ( const key in replacementObj ) {
+			if ( replacementObj.hasOwnProperty( key ) ) {
```

Comment:
Can you reduce the nesting? Extract to smaller functions?

		for ( const key in replacementObj ) {
		  deepMergeKey( ...... );
		}

in  deepMergeKey 
if (replacementObj.hasOwnProperty( key){
return 
}